### ISSUE TITLE
Requesting Assistance - "Make Run"

### ISSUE THREAD

**Issue by digitalfaux** [pilsnerbeer/multisig-tool-fixed#3]
Hi,

I have done npm install and running localhost:8000 on python3. Site loads everything up except in the command prompt it says 404 the "multisig.js" not found. The continue buttons don't work or anything.

I'm also not understanding the "make run" part. I just don't know what this means at all. Does anyone know if I'm missing a step somewhere?

**Comment by miroslav-brewmaster** [comment]
Hi,

It seems you have missed an important step of the installation process. The `make run` command (if you look at Makefile in repo) excutes `compile` , which uses browserify to bundle all the necessary code into multisig.js. Without it the app will not run.

You need GNU Make in order to run this command (https://www.gnu.org/software/make/). It will run on Linux and MacOS, on Windows you can use WSL. Alternatively, you could execute the `compile` script without needing to use Make at all:

```
	cd lib/multisig && \
	../../node_modules/.bin/browserify -r ./vault -r ./views/addresses_view > ../../multisig.js 
```
(after this run the server.)


Hope that helped you.


PS if you can't really get this to work I could precompile the project & upload it to releases but hopefully you can manage :-)

**Comment by pilsnerbeer** [comment]
Hi,

It seems you have missed an important step of the installation process. The make run command (if you look at Makefile in repo) excutes compile , which uses browserify to bundle all the necessary code into multisig.js. Without it the app will not run.

You need GNU Make in order to run this command (https://www.gnu.org/software/make/). It will run on Linux and MacOS, on Windows you can use WSL. Alternatively, you could execute the compile script without needing to use Make at all:

	cd lib/multisig && \
	../../node_modules/.bin/browserify -r ./vault -r ./views/addresses_view > ../../multisig.js 
(after this run the server.)

Hope that helped you.

PS if you can't really get this to work I could precompile the project & upload it to releases but hopefully you can manage :-)

**Comment by digitalfaux** [comment]
> Hi,
> 
> It seems you have missed an important step of the installation process. The make run command (if you look at Makefile in repo) excutes compile , which uses browserify to bundle all the necessary code into multisig.js. Without it the app will not run.
> 
> You need GNU Make in order to run this command (https://www.gnu.org/software/make/). It will run on Linux and MacOS, on Windows you can use WSL. Alternatively, you could execute the compile script without needing to use Make at all:
> 
> ```
> cd lib/multisig && \
> ../../node_modules/.bin/browserify -r ./vault -r ./views/addresses_view > ../../multisig.js 
> ```
> 
> (after this run the server.)
> 
> Hope that helped you.
> 
> PS if you can't really get this to work I could precompile the project & upload it to releases but hopefully you can manage :-)

I got Make installed. I Make Run for the directory and it creates the multisig.js but I believe it's failing.
![Make Run](https://github.com/pilsnerbeer/multisig-tool-fixed/assets/139646510/94cb8688-ed3c-47e4-a2fc-1d35aed2d8fa)

**Comment by pilsnerbeer** [comment]
I was able to run this with no issues on MacOS using Make 3.81. Maybe there is an issue with command chaining in cmd/powershell (are you on windows?). You can still execute the browserify command manually OR use multisig.js I just uploaded to Releases which contains all necessary JS code. You can drop it into the root directory and run `python3 -m http.server` and everything should start working.

**Comment by digitalfaux** [comment]
Thanks for your help. Everything looks like it works now with the compiled multisig.js you provided. I've double checked my keys multiple times and it seems the tool just hangs for along time at step 1. I left it alone for a couple hours multiple times trying. I've just been unsuccessful. Oh well, my $13 of BTC will be stuck in the coinbase multisig until the end of time annoying me every time I look at my balance that's unavailable lol.

Hopefully coinbase can just disconnect/disassociate the wallet from my account.

**Comment by pilsnerbeer** [comment]
sorry to hear that,addresses are normally found within seconds. You could check browser console and see what exactly the error is.

**Comment by digitalfaux** [comment]
![image](https://github.com/pilsnerbeer/multisig-tool-fixed/assets/139646510/0312aae5-e50a-4ebe-b73f-bafed4ec5752)

This is the error I got. Unfortunately it's gibberish to me. I imagine something is up with the multisig file.

**Comment by pilsnerbeer** [comment]
If everything runs fine I think the keys you are inputting are incorrect. Double-check that they are in the right fields and that you have copied them correctly. the tool has been tested multiple times, so I expect there is something wrong with the keys not the app
